sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -179,13 +179,17 @@ static Context mergeRequestOptionsContext(Context context, RequestOptions option
     }
 
     static Flux<ByteBuffer> validateLength(final HttpRequest request) {
-        final Flux<ByteBuffer> bbFlux = request.getBody();
-        if (bbFlux == null) {
+        if (request.getBody() == null) {
             return Flux.empty();
         }
 
-        final long expectedLength = Long.parseLong(request.getHeaders().getValue("Content-Length"));
+        final Flux<ByteBuffer> bbFlux = request.getBody()
```

Comment:
Mind adding a test to this that ensures a replayable stream that is subscribed to twice doesn't result in an error being thrown, which it is trying to resolve.